lib/pdk/generate/module.rb

```diff
@@ -135,13 +135,13 @@ def self.prepare_metadata(opts = {})
         require 'pdk/answer_file'
         require 'pdk/module/metadata'
 
-        opts[:username] = (opts[:username] || PDK.answers['forge_username'] || username_from_login).downcase
+        opts[:username] = (opts[:username] || PDK.config.get_within_scopes('module_defaults.forge_username') || username_from_login).downcase
 
         defaults = PDK::Module::Metadata::DEFAULTS.dup
 
         defaults['name'] = "#{opts[:username]}-#{opts[:module_name]}" unless opts[:module_name].nil?
-        defaults['author'] = PDK.answers['author'] unless PDK.answers['author'].nil?
-        defaults['license'] = PDK.answers['license'] unless PDK.answers['license'].nil?
+        defaults['author'] = PDK.config.get_within_scopes('module_defaults.author') unless PDK.config.get_within_scopes('module_defaults.author').nil?
```

Comment:
We should come up with a helper around this get-and-set-unless-nil pattern that isn't so verbose. Maybe a method that yields the value to a block if not nil?

Something like
```
PDK.config.with_scoped_value('module_defaults.author') { |v| defaults['author'] = v }
```

Reply:
I like that!  Will implement.

Reply:
Added a commit which does this.